src/calculators/SVCalculator.cpp

```diff
@@ -0,0 +1,86 @@
+#include "SVCalculator.h"
+
+#include <cmath>
+
+namespace calculators {
+
+    void SVCalculator::calculate(ParticleContainer &particleContainer, double delta_t) {
+        calculateX(particleContainer, delta_t);
+        calculateF(particleContainer);
+        calculateV(particleContainer, delta_t);
+    }
+
+    void SVCalculator::calculateF(ParticleContainer &particleContainer) {
+        for (auto p = particleContainer.begin(); p != particleContainer.end(); ++p) {
+            p->setOldF(p->getF());  // Update oldF with currentF
+            p->setF({0, 0, 0});     // Reset F to zeros
+        }
+
+        // Iterate over all unique pairs of particles
+        for (auto pair = particleContainer.pair_begin(); pair != particleContainer.pair_end(); ++pair) {
+            // Get the positions and masses of the two particles
+            std::array<double, 3> x1 = pair->first.get().getX();
+            std::array<double, 3> x2 = pair->second.get().getX();
+
+            double m1 = pair->first.get().getM();
+            double m2 = pair->second.get().getM();
+
+            // Calculate the distance vector and its norm
+            std::array<double, 3> dx = {x2[0] - x1[0], x2[1] - x1[1], x2[2] - x1[2]};
+            double distance = std::sqrt(dx[0] * dx[0] + dx[1] * dx[1] + dx[2] * dx[2]);
```

Comment:
You could mark all of these intermediate values as `const`. That makes it easier to see what variables are meant to change and which are just constant values.
Sometimes, also helps the compiler to optimize stuff.